include/forcing/ForcingsEngineLumpedDataProvider.hpp

```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include "DataProviderSelectors.hpp"
+#include "ForcingsEngineDataProvider.hpp"
+
+#include <utilities/mdarray.hpp>
+
+namespace data_access {
+
+struct ForcingsEngineLumpedDataProvider
+  : public ForcingsEngineDataProvider<double, CatchmentAggrDataSelector>
+{
+    static constexpr auto bad_index = static_cast<std::size_t>(-1);
+
+    ~ForcingsEngineLumpedDataProvider() override = default;
+
+    double get_value(const CatchmentAggrDataSelector& selector, data_access::ReSampleMethod m) override;
+
+    std::vector<double> get_values(const CatchmentAggrDataSelector& selector, data_access::ReSampleMethod m) override;
+
+    /**
+     * @brief Get the index in `CAT-ID` for a given divide in the instance cache.
+     * @note The `CAT-ID` output variable uses integer values instead of strings.
+     * 
+     * @param divide_id A hydrofabric divide ID, i.e. "cat-*"
+     * @return size_type 
+     */
+    std::size_t divide_index(const std::string& divide_id) noexcept;
+
+    /**
+     * @brief Get the index of a variable in the instance cache.
+     * 
+     * @param variable 
+     * @return size_type 
+     */
+    std::size_t variable_index(const std::string& variable) noexcept;
+
+    static ForcingsEngineDataProvider* lumped_instance(
+        const std::string& init,
+        const std::string& time_start,
+        const std::string& time_end,
+        const std::string& time_fmt = default_time_format
+    );
+
+  private:
+    ForcingsEngineLumpedDataProvider(
+        const std::string& init,
+        std::size_t time_begin_seconds,
+        std::size_t time_end_seconds
+    );
+
+    /**
+     * @brief Update to next timestep.
+     * 
+     * @return true
+     * @return false 
+     */
+    bool next();
+
+    /**
+     * @brief Get a forcing value from the instance
+     * 
+     * @param divide_id Divide ID to index at
+     * @param variable Forcings variable to get
+     * @param previous If true, return the previous timestep values.
+     * @return double 
+     */
+    double at(
+        const std::string& divide_id,
+        const std::string& variable,
+        bool previous = false
```

Comment:
So, I feel rather late pointing this out, but `at()` is private, defaults `previous = false`, and has no internal callers for which it's passed an explicit `true`. At that point, what are we getting for doubling our memory footprint and the work of double-buffering the data in that memory?

Reply:
Initially I thought that requiring the previous timestep values would be necessary so that catchments (in the same process) referencing the same forcings engine instance did not pass-over a step needed by other catchments -- but looking back through the main NGen code, I think it was a premature implementation and won't be needed -- so I will take out that portion of this PR